File: Day<10>/Solution.py
class Solution:
    def repeatedStringMatch(self, A,B):
        S=A
        ret=1
        
        while len(S)<len(B):
            S+=A
            ret+=1
        
        if B in S:
            return ret

        if B in S+A:
            return ret+1
        return -1

if __name__=="__main__":
    t=int(input())
    for _ in range(t):
        A=input().strip()
        B=input().strip()
        obj = Solution()
        print(obj.repeatedStringMatch(A,B))

File: Day<10>/Solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution
{
    private:
    bool isSubString(string s, string sub)
    {
        int count = 0;
        for (int i = 0; i < s.length(); i++)
        {
            if (s[i] == sub[count])
                count++;
            else
                i-=count;
                count = 0;
            if (count == sub.length())
                return true;
        }
        return false;
    }

    public:
    int repeatedStringMatch(string A, string B)
    {
        string temp = A;
        int count = ceil(float(B.length()) / float(A.length()));

        for (int i=1; i<count; i++)
            A += temp;

        if (isSubString(A,B))
            return count;
        else
        {
            A += A;
            if (isSubString(A,B))
                return count+1;
        }
        return -1;
    }
};

int main()
{
    int t;
    cin>>t;
    while (t--)
    {
        string A,B;
        cin>>A;
        cin>>B;
        Solution obj;
        cout<<obj.repeatedStringMatch(A,B)<<"\n";
    }
    return 0;
}
